src/arena/gate.rs

```diff
@@ -57,44 +58,36 @@ pub fn despawn_gates_system(
                 };
 
                 // verify the other entity is a spawnable
-                if spawnable_query
-                    .iter()
-                    .any(|spawnable_entity| spawnable_entity == *other_entity)
-                {
+                if spawnable_query.contains(*other_entity) {
                     // despawn the spawnable entity
                     commands.entity(*other_entity).despawn_recursive();
 
                     // check if the other entity is a mob
-                    for (mob_entity, mob_component) in mob_query.iter() {
-                        if mob_entity == *other_entity {
-                            // send event for mob reaching bottom of arena
-                            if let Some(defense_interaction) =
-                                mob_component.defense_interaction.clone()
-                            {
-                                enemy_bottom_event.send(MobReachedBottomGateEvent {
-                                    mob_type: Some(mob_component.mob_type.clone()),
-                                    mob_segment_type: None,
-                                    defense_interaction,
-                                });
-                            }
+                    if let Ok((_, mob_component)) = mob_query.get(*other_entity) {
```

Comment:
Dang i should have made this a separate commit, but im done with `git add -p` for today.....